FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenExternalIntentsFlowTest.kt

```diff
@@ -460,20 +461,20 @@ class ComposeScreenExternalIntentsFlowTest : BaseTest() {
   }
 
   private fun checkAtts(attachmentsCount: Int) {
-    onView(withId(R.id.layoutAtts))
-      .check(matches(hasChildCount(attachmentsCount)))
-
-    when {
-      attachmentsCount > 0 -> {
-        if (attachmentsCount == 1) {
-          onView(withText(atts.first().name))
-            .check(matches(isDisplayed()))
-        } else {
-          for (att in atts) {
-            onView(withText(att.name))
-              .check(matches(isDisplayed()))
-          }
-        }
+    onView(withId(R.id.rVAttachments))
+      .check(matches(withRecyclerViewItemCount(attachmentsCount)))
+
+    if (attachmentsCount > 0) {
+      for (i in 0 until attachmentsCount) {
```

Comment:
how about using `forEach`?
```kt
atts.forEach { attachment ->
}
```

Reply:
Or there're some other logic behind this?

Reply:
`0 until attachmentsCount` - I don't do iteration over the whole list(`atts`). `atts` contains 3 items, but `attachmentsCount` can be different.

Reply:
Also I prefer to use take with foreach in this case like below.
But if you prefer to use `for` `in`, you may leave it like it
```kt
atts.take(attachmentsCount).forEach { attachment ->
```

Reply:
> Makes sense. Can attachmentsCount be minus value?
I'm asking because I think we won't need above if statement

Agree. `attachmentsCount` can be minus value by mistake. But anyway, a test will be completed only for a positive value(> 0)

Reply:
> Also I prefer to use take with foreach in this case like below.

Nice. It looks shorter. I like it.